module/validation/receipt_validator.go

```diff
@@ -272,12 +277,28 @@ func (v *receiptValidator) ValidatePayload(candidate *flow.Block) error {
 		return fmt.Errorf("internal error while traversing the ancestor fork of unsealed blocks: %w", err)
 	}
 
+	// tracks the number of receipts committing to each result.
+	// it's ok to only index receipts at this point, because we will perform
+	// all needed checks after we have validated all results.
+	receiptsByResult := payload.Receipts.GroupByResultID()
+
 	// first validate all results that were included into payload
-	// if one of results is invalid we fail the whole check because it could be violating
-	// parent-children relationship
+	// if one of results is invalid we fail the whole check because it could be violating parent-children relationship
+	// each execution
 	for i, result := range payload.Results {
 		resultID := result.ID()
 
+		// check if there are enough execution receipts included in the payload corresponding to the execution result
+		receiptsForResult := uint(len(receiptsByResult.GetGroup(resultID)))
+		if receiptsForResult > 0 {
+			if receiptsForResult < v.requiredReceiptsIncludedForExecutionResult {
+				return engine.NewInvalidInputErrorf("execution result %v has only %d receipts, but at least %d are required",
+					resultID, receiptsForResult, v.requiredReceiptsIncludedForExecutionResult)
+			}
+		} else {
```

Comment:
```suggestion
		if receiptsForResult == 0 {
```

Reply:
agree with Jordan:
for each result that is incorporated in a block, there must be at least one receipt incorporated in the same block.